.github/workflows/ghcr-publish-snapshot.yaml

```diff
@@ -0,0 +1,68 @@
+# This file is part of Astarte.
+#
+# Copyright 2025 SECO Mind Srl
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: Publish snapshot Docker images
+on:
+  workflow_dispatch:
+  workflow_call:
+env:
+  IMAGE: ghcr.io/${{ github.repository_owner }}/astarte-message-hub
+jobs:
+  push_snapshot_to_registry:
+    name: Push Docker images to registry
+    runs-on: ubuntu-22.04
+    permissions:
+      actions: write
+      packages: write
+      contents: read
+      attestations: write
+      id-token: write
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v4
+      - name: Check image file changes
+        uses: dorny/paths-filter@v3
+        id: changes
+        with:
+          filters: |
+            tf:
+              - 'src/**'
+              - '.github/workflows/ghcr-publish-snapshot.yaml'
+              - 'Dockerfile'
+      - name: Skip execution if no changes to image files
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        if: steps.changes.outputs.tf != 'true'
+        run: |
+          gh run cancel ${{ github.run_id }}
+          gh run watch ${{ github.run_id }}
```

Comment:
I would always publish the snapshot. The file filter is missing the Cargo.toml for example. I think we can stay on the safer side and always publish the snapshot